File: Day 1/1/reverseString.js
const reverseString = (str) => {
    return str.split("").reverse().join("");
}

const reverseStringAlgorithmically = (str) => {
    let reversed = "";
    
    for (let i = str.length - 1; i >= 0; i--) {
        reversed += str[i];
    }

    return reversed;
}


console.log(reverseString("hello"));
console.log(reverseStringAlgorithmically("hello"));

File: Day 1/1/reverse_string.c
#include <stdio.h>

void reverseString(char str[]) {
    int length = 0;
    
    while (str[length] != '\0') {
        length++;
    }

    for (int i = 0; i < length / 2; i++) {
        char temp = str[i];
        str[i] = str[length - 1 - i];
        str[length - 1 - i] = temp;
    }
}

int main() {
    char string[] = "Hello World";

    printf("Original string: %s\n", string);

    reverseString(string);

    printf("Reversed string: %s\n", string);

    return 0;
}
